pkg/connectivity/pod-to-pod.go

```diff
@@ -22,3 +25,12 @@ func PodToPod(fromPod *v1.Pod, toPod *v1.Pod) common.Result {
 		Successful: false,
 	}
 }
+
+func hasEnvoySideCar(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		if strings.Contains(container.Image, "envoy") {
+			return true
+		}
+	}
+	return false
```

Comment:
Checking the image name does not correct. One could easily create a container image with the string `envoy`, which doesn't imply there is a sidecar present.

Reply:
Would it make more sense to check for the image of the container to be something like `Image:         mcr.microsoft.com/oss/envoyproxy/envoy:...`

Reply:
No, the OSM repo is not aware of this image and does not maintain it. The image name could be made configurable, with the default pointing to the image specified in https://github.com/openservicemesh/osm/blob/main/charts/osm/values.yaml#L20

Reply:
We can also check for the envoy unique id label right? Pods that have this will be a part of the mesh and have envoy sidecars.

```
	listOptions := metav1.ListOptions{
		// Matches on pods which are already a part of the mesh, which contain the Envoy ID label
		LabelSelector: constants.EnvoyUniqueIDLabelName,
	}
```

Reply:
Not necessarily it seems? I removed a ns from the mesh and its pod still had a `osm-proxy-uuid` uuid. That being said, it does still have the envoy sidecar as well soooo depends what we're trying to do with this information
> Pods that have this will be a part of the mesh

Reply:
Is this expected behavior....?

Reply:
hasEnvoySidecar() needs to include a few checks, all of which are not going to ensure 100% correctness, but should suffice to start with:
1. verify the pod is meshed. This can be identified by inspecting the pod's label, and ensuring it has at least 3 containers (app, init, sidecar)
2. Verify that the non-app containers correspond to 1. osm-init, 2. envoy

Reply:
You could get the specific envoy image from meshconfig